lib/nerves_hub/archives.ex

```diff
@@ -66,6 +67,24 @@ defmodule NervesHub.Archives do
     |> Repo.one!()
   end
 
+  @spec archive_for_deployment(integer()) :: Archive.t() | nil
+  def archive_for_deployment(nil), do: nil
+
+  def archive_for_deployment(deployment_id) do
+    deployment =
+      Deployment
+      |> where(id: ^deployment_id)
+      |> join(:inner, [d], a in assoc(d, :archive))
+      |> preload([_, a], archive: a)
+      |> Repo.one()
+
+    if deployment do
+      deployment.archive
+    else
+      nil
+    end
```

Comment:
Would it make sense to query from the Archive so that is the return instead of having to do a check?
```suggestion
    Archive
    |> join(:inner, [a], d in Deployment, on: d.archive_id == a.id)
    |> where([a, d], d.id == ^deployment_id)
    |> Repo.one()
```